### PR TITLE
[4.0] Fix for Synchronization of module filter when changing admin and site

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for Issue # https://github.com/joomla/joomla-cms/issues/30302.

### Summary of Changes
Add a GET-Variable for determining if client has changed. 
If the filters are used, the form is submitted and the GET-variable is set. If the client changes via the menu or the Select, it is only a redirected.

### Testing Instructions
See https://github.com/joomla/joomla-cms/issues/30302.

### Actual result BEFORE applying this Pull Request
See https://github.com/joomla/joomla-cms/issues/30302.

### Expected result AFTER applying this Pull Request
Filters are completely in default state (not only in view) if client changed.

### Documentation Changes Required

No

### CODE REVIEW

**Comment by richard67** [comment]
@astridx Why sometimes double quotes `"`and sometimes single quotes `'` for strings? As far as I know we shall use single quotes.

File: administrator/components/com_modules/src/Model/ModulesModel.php [hunk 1]
```diff
@@ -87,12 +87,17 @@ protected function populateState($ordering = 'a.position', $direction = 'asc')
 			$this->context .= '.' . $layout;
 		}
 
-		// Load the filter state.
-		$this->setState('filter.search', $this->getUserStateFromRequest($this->context . '.filter.search', 'filter_search', '', 'string'));
-		$this->setState('filter.position', $this->getUserStateFromRequest($this->context . '.filter.position', 'filter_position', '', 'string'));
-		$this->setState('filter.module', $this->getUserStateFromRequest($this->context . '.filter.module', 'filter_module', '', 'string'));
-		$this->setState('filter.menuitem', $this->getUserStateFromRequest($this->context . '.filter.menuitem', 'filter_menuitem', '', 'cmd'));
-		$this->setState('filter.access', $this->getUserStateFromRequest($this->context . '.filter.access', 'filter_access', '', 'cmd'));
+		$clientNotChanged = $app->input->get('client_not_changed', "0", 'cmd');
```
Comment (richard67) [inline administrator/components/com_modules/src/Model/ModulesModel.php]:
```suggestion
		$clientNotChanged = $app->input->get('client_not_changed', '0', 'cmd');
```

File: administrator/components/com_modules/src/Model/ModulesModel.php [hunk 2]
```diff
@@ -87,12 +87,17 @@ protected function populateState($ordering = 'a.position', $direction = 'asc')
...
+
+		if ($clientNotChanged === "1")
```
Comment (richard67) [inline administrator/components/com_modules/src/Model/ModulesModel.php]:
```suggestion
		if ($clientNotChanged === '1')
```

File: administrator/components/com_modules/src/Model/ModulesModel.php [hunk 3]
```diff
@@ -131,7 +136,10 @@ protected function populateState($ordering = 'a.position', $direction = 'asc')
 		$this->setState('params', $params);
 
 		// List state information.
-		parent::populateState($ordering, $direction);
+		if ($clientNotChanged === "1")
```
Comment (richard67) [inline administrator/components/com_modules/src/Model/ModulesModel.php]:
```suggestion
		if ($clientNotChanged === '1')
```

**Comment by richard67** [comment]
@astridx Could you check and apply my above suggested changes? I could give your PR a good test then because I have already tested it with these changes.

**Comment by Quy** [comment]
The filter is collapsed upon switching, however, `cpanel` is still selected when filter is expanded, but not filtered.

**Comment by ceford** [comment]
Confirming what Quy said. Also, if I select a Site Position filter and switch back forth between Site and Administrator then the Site filter is still selected but the list is not filtered.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30323">issues.joomla.org/tracker/joomla-cms/30323</a>.</sub>

**Comment by astridx** [comment]
I thought I had a fix.
But this was bad and now I see, that  joomla.resetfilter

https://github.com/joomla/joomla-cms/blob/83742b2e18e54da25c4e64e1f913496afa391115/build/media_source/system/js/searchtools.es6.js#L13 
does not work.

Not only in com_modules, in no component, or do I oversee something? In other components it is not so obviously.


So this need another fix and I close here.

Can someone reopen
Issue  #30302 please.

**Comment by richard67** [comment]
> Can someone reopen
> Issue  #30302 please.

Done.

**Pull request closed without merging**